pinconfigs/README.md

```diff
@@ -0,0 +1,154 @@
+devices using the *AF model*.
+
+In the following sections the expected file format is described.
+
+## Common configuration
+
+Some fields are common independently of the pin configuration model.
+The commented example below describes common fields in detail.
+
+- `model` (required): Choose between `afio` or `af`
+- `series` (required): Series name, e.g. gd32vf103
+- `variants` (required): Each variant has a different set of valid pin
+  combinations, either because of a different number of pins or because devices
+  with certain memory configurations have less or more peripherals available.
+
+Example:
+
+```yaml
+model: afio
+
+series: gd32vf103
+
+variants:
+  - pincode: V
+    memories: [B, 8]
+  - pincode: R
+    memories: [B, 8]
+  # Same pin code as before but different memories: a variant because it has
+  # less peripherals (and so less signals) than B-8.
+  - pincode: R
+    memories: [6, 4]
+```
+
+## AFIO model
+
+The AFIO model supports the following fields in the configuration file:
+
+- `signal-configs` (required): A dictionary of signal configurations. Each
+  available signal is required since the mode has to be specified.
+
+  Available configuration fields for each signal are:
+
+  - `modes` (required): A list containing one or more of these modes: `analog`,
+    `inp` and `out`.
+  - `exclude-memories` (optional): A lit of memories where the signal is not
+    available. If not defined, it is assumed that signal is available on all
+    memory configurations.
+  - `exclude-pincodes` (optional): A list of pincodes where the signal is not
+    available. If not defined, it is assumed that signal is available on all
+    pin codes.
+
+- `pins` (required): A dictionary of pin configurations.
+
+  Available configuration options for each pin are:
+
+  - `pincodes` (required): A list of pin codes that contain this pin.
+  - `afs` (required): The list of valid alternate functions for that pin.
+    The list must not include signals that can be remapped.
+
+- `remaps` (required): Available remaps for each remappable signal.
+
+  Available configuration options for each remap are:
+
+  - `pins`: Available pins. The size of the list of pins determines wether
+    a signal can be remapped (2 options) or partially or fully remapped
+    (4 options). If a signal has a single partial remap, it still needs
+    to provide 4 entries, but setting the non-available one to null.
+
+Example:
+
+```yaml
+signal-configs:
+  # Configuration for 'ADC01_IN0': available on all pincodes/memories, can
+  # only operate in 'analog' mode.
+  ADC01_IN0:
+    modes: [analog]
+  # Configuration for 'I2S2_CK': available on all pincodes, all memories except
+  # 6, 4 and can operate in both 'inp' and 'out' modes.
+  I2S2_CK:
+    modes: [inp, out]
+    exclude-memories: [6, 4]
+
+pins:
+  # Configuration for pin 'PA0'. Supported on V, R, C, T pincodes, valid
+  # alternate functions include WKUP, USART1_CTS, ...
+  PA0:
+    pincodes: [V, R, C, T]
```

Comment:
For GD32F405/407, BGA100 and LQPF100 both have the 'V' pincode, but BGA100 doesn't have the PD8 pin. It can be check at datasheet 2.3, 2.6 pins related section. GD32F403 doesn't have this issue.

I think it's a GigaDevice mistake, they told GD32F405/407 BGA100 have 82 IO too, but actually is 81.